modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.pagememory.persistence.store;
+
+import static org.apache.ignite.internal.pagememory.persistence.store.FilePageStoreHeader.readHeader;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.ignite.internal.fileio.FileIo;
+import org.apache.ignite.internal.fileio.FileIoFactory;
+import org.apache.ignite.lang.IgniteInternalCheckedException;
+
+/**
+ * Factory for creating {@link FilePageStore}.
+ */
+class FilePageStoreFactory {
+    /** Latest file page store version. */
+    private final int latestVersion = FilePageStore.VERSION_1;
+
+    private final FileIoFactory fileIoFactory;
+
+    private final int pageSize;
+
+    /**
+     * Constructor.
+     *
+     * @param fileIoFactory File IO factory.
+     * @param pageSize Page size in bytes.
+     */
+    public FilePageStoreFactory(FileIoFactory fileIoFactory, int pageSize) {
+        this.fileIoFactory = fileIoFactory;
+        this.pageSize = pageSize;
+    }
+
+    /**
+     * Creates instance of {@link FilePageStore}.
+     *
+     * <p>If the file exists, an attempt will be reads its {@link FilePageStoreHeader header} and create the {@link FilePageStore}.
```

Comment:
What did you try to say with "an attempt will be reads its"? I don't understand it

Reply:
My english is bad. Wanted to say:
If the file exists (not empty), then at the beginning it will read the header and create a FilePageStore based on it.

Reply:
It's not that bad, I guess that current line was born out of several conflicting sentences and you forgot to clean it up.
I think it supposed to be "... an attempt will be **made to read** its ..."

Reply:
fix it